### ISSUE TITLE
kmod: fails to build on aarch64-darwin

### ISSUE THREAD

**Issue by Sciencentistguy** [NixOS/nixpkgs#184213]
### Steps To Reproduce
1. build `kmod`

### Build log
```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/islh0pxjqwqky6a3syraxxycxr19rj8c-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
applying patch /nix/store/i4q069m3d29gi7dw1ijhjrp4hxw1c2l4-module-dir.patch
patching file Makefile.am
patching file configure.ac
Hunk #1 succeeded at 276 with fuzz 2 (offset 74 lines).
patching file libkmod/libkmod.c
Hunk #1 succeeded at 208 (offset 2 lines).
Hunk #2 succeeded at 224 (offset 2 lines).
patching file tools/static-nodes.c
applying patch /nix/store/k6h6xgq4440gp3gwn3wbg4nqbggnd24q-darwin.patch
patching file Makefile.am
Hunk #1 succeeded at 99 (offset 19 lines).
Hunk #2 succeeded at 109 (offset 19 lines).
patching file libkmod/libkmod-module.c
Hunk #1 succeeded at 849 with fuzz 1 (offset 62 lines).
Hunk #2 succeeded at 946 (offset 63 lines).
patching file libkmod/libkmod-signature.c
Hunk #1 succeeded at 17 with fuzz 2.
patching file shared/macro.h
patching file shared/missing.h
patching file shared/util.c
Hunk #1 succeeded at 370 (offset 3 lines).
patching file shared/util.h
Hunk #1 FAILED at 7.
1 out of 1 hunk FAILED -- saving rejects to file shared/util.h.rej

```

### Additional context
Problem started with Commit [3feca2a](https://github.com/NixOS/nixpkgs/commit/3feca2a040927cffcb78a47598b92e390f0d7b68) 

### Notify maintainers
@Artturin 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.5.0, macOS 12.4`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.10.3`
 - channels(jamie): `""`
 - channels(root): `""`
 - nixpkgs: `/etc/nix/path/nixpkgs`
```

**Comment by wegank** [comment]
This seems to be intended, see #182714.

**Comment by Sciencentistguy** [comment]
Well, this (transitively) breaks neofetch on aarch64-darwin

**Comment by wegank** [comment]
It is already fixed in staging (#182720). Still have to wait a few days.

**Comment by Sciencentistguy** [comment]
👍
